### PR TITLE
feather-stm32f405: add UART support

### PR DESCRIPTION
**Pull request by ardnew**
This adds support for the three UART peripherals on the target `feather-stm32f405`.

### CODE REVIEW

File: src/machine/board_feather-stm32f405.go [hunk 1]
```diff
@@ -113,7 +118,30 @@ const (
...
+		Bus:             stm32.USART1,
+		AltFuncSelector: stm32.AF7_USART1_2_3,
+	}
+	UART0 = UART1
+)
+
+func initUART() {
+	UART1.Interrupt = interrupt.New(stm32.IRQ_USART3, UART1.handleInterrupt)
+	UART2.Interrupt = interrupt.New(stm32.IRQ_USART6, UART2.handleInterrupt)
+	UART3.Interrupt = interrupt.New(stm32.IRQ_USART1, UART3.handleInterrupt)
```
Comment (ardnew) [inline src/machine/board_feather-stm32f405.go:143]: Should we be automatically installing these interrupts for the non-default UARTs (`UART2`/`UART3`)? The default `UART1` is the only one required by other packages (e.g. `runtime.putchar()`)

The other UARTs will function with this left as-is, but it's more of a question of convention and how other targets provide extra peripherals to the user

Comment (deadprogram) [reply src/machine/board_feather-stm32f405.go:143]: > Should we be automatically installing these interrupts for the non-default UARTs 

Possibly not? If a developer wants to use them for a specific application, they can do that themselves.

Comment (aykevl) [reply src/machine/board_feather-stm32f405.go:143]: Probably yes. This is a bit of magic code that gets optimized away if the UART is not in use (so no interrupt should be registered if you don't use UART2 for example).

**Comment by aykevl** [comment]
The code itself looks fine.

**Comment by deadprogram** [comment]
Thanks for working on this contribution @ardnew now merging.

**Pull request merged**